kiwi/bootloader/config/grub2.py

```diff
@@ -657,6 +659,14 @@ def _setup_default_grub(self):
             r'(^root=[^\s]+)|( root=[^\s]+)', '', self.cmdline
         ).strip()
         if self.persistency_type == 'by-label':
+            label = re.search(r'(^root=[^\s]+)|( root=[^\s]+)', self.cmdline)
+            if label and label.groups()[-1]:
+                label_definition = label.groups()[-1].strip()
+                if 'LABEL' in label_definition:
+                    grub_default_entries['GRUB_CMDLINE_LINUX'] = \
+                        '"{0}"'.format(label_definition)
```

Comment:
Do you have a specific reason why the ```root=LABEL=foo``` is added to the GRUB_CMDLINE_LINUX variable instead of the GRUB_CMDLINE_LINUX_DEFAULT variable ?

Am I correct that writing the ```root=``` information will disable the grub tooling to operate on the root device and simply trust the value given here ?

If so this would invalidate the setting GRUB_ENABLE_LINUX_LABEL to be effective.

Would it be possible that your code changes only apply if GRUB_ENABLE_LINUX_LABEL is not available ?
I think the scope should be limited. If you look in ```bootloader/config/grub2.py``` you see examples for which we only handle the grub option if there is an implementation for it available. e.g ```enable_blscfg_implemented = ...```

Thanks